### ISSUE TITLE
Is there a way to check if your content is going to throw an error?

### ISSUE THREAD

**Issue by hahnbeelee** [hashicorp/next-mdx-remote#310]
I'm building an application using `next-mdx-remote` that has users who will write up custom content/pages. It isn't guaranteed that the content is always going to be correct/compile. This is what I'm doing right now:
```tsx
export default function Page({ stringifiedMdxSource }: PageProps) {
  const mdxSource = JSON.parse(stringifiedMdxSource);
  return <MDXRemote components={components} {...mdxSource} />
}
```
Is there a way for me to check if the content will compile? I don't want my entire application to break due to one person uploading one incorrect file. I want it to silently fail. Thank you!

**Comment by BRKalow** [comment]
Most errors that would happen as a result of malformed content would be thrown from the call to `serialize()`, which should be happening on the server. Wrapping the serialize call in a `try/catch` would be a potential solution. Can you share more details about your setup?

**Comment by hahnbeelee** [comment]
Most of the time my serialize function is fine. The reason why I'm specifically talking about `<MDXRemote components={components} {...mdxSource} />` is because sometimes components are used when they aren't a part of the components I fed into MDXRemote. This is [where I'm using MDXRemote](https://github.com/mintlify/mint/blob/vercel-platforms-migration/client/src/pages/_sites/%5Bsite%5D/%5B%5B...slug%5D%5D.tsx#L186) and this is where I'm using [serialize](https://github.com/mintlify/mint/blob/vercel-platforms-migration/client/src/utils/mdx/getMdxSource.ts). I'm thinking of using [Error Boundaries ](https://nextjs.org/docs/advanced-features/error-handling#handling-client-errors) but it hasn't 100% been successful yet, so I'm all ears if you have any recommendations.

**Comment by BRKalow** [comment]
Makes sense! Is all of the content rendered at build-time and are the components you pass into `MDXRemote` a fixed list? If so, I personally would consider using a remark plugin (maybe [remark-lint](https://github.com/remarkjs/remark-lint)) to check that all components being used are declared.

Catching this during `serialize()` would be a faster feedback cycle for authors.

**Comment by hahnbeelee** [comment]
I've added try catches and created a remark plugin to remove all components that aren't in my list. Unfortunately the try catch around the serialize function, nor the try catch around my content are catching syntax errors.
<img width="1274" alt="image" src="https://user-images.githubusercontent.com/55263191/199778727-5d42669f-4de5-4276-993e-f9b74369ab9c.png">

[Try catch around MDXRemote](https://github.com/mintlify/mint/blob/vercel-platforms-migration/client/src/pages/_sites/%5Bsubdomain%5D/%5B%5B...slug%5D%5D.tsx#L45-L80)
[Try catch around serialize](https://github.com/mintlify/mint/blob/vercel-platforms-migration/client/src/pages/_sites/%5Bsubdomain%5D/%5B%5B...slug%5D%5D.tsx#L142-L148)

Any ideas for catching compilation errors outside of using a try catch?

**Comment by BRKalow** [comment]
What other syntax errors are you looking to catch? Do you have an example of something invalid that's not being handled? It looks like you've covered undefined components with that remark plugin.

**Comment by hahnbeelee** [comment]
The most common ones I see:
- Sometime people write <user_id> when talking about a variable but MDX detects it as a component (and is broken because it doesn't have a closing component </user_id>)
- `<` and `{` need to be escaped (`/<` `/{`)

These just cause the MDX to not parse. It's really going over my head why my try catch during serialize, nor my try catch around my MDXRemote aren't catching them.

I know the easy solution is just fixing these typos but it's really important that my whole application doesn't break because of a syntax error one user made.

**Comment by hahnbeelee** [comment]
I just found [this issue](https://github.com/mdx-js/mdx/discussions/1862) I've been putting error boundaries, but they are also not being triggered. Curious if you think that MDXRemote wraps MDXProvider in a way that would make the error boundary not work.

**Comment by BRKalow** [comment]
> The most common ones I see:
> 
> Sometime people write <user_id> when talking about a variable but MDX detects it as a component (and is broken because it doesn't have a closing component </user_id>)
> < and { need to be escaped (/< /{)

Based on the image you shared and the code you linked, these types of things are being caught by the `try/catch` around `serialize()` 🤔 

Here's a test case: https://github.com/hashicorp/next-mdx-remote/blob/f5b0e74529908efd78b981bae7121847ed751b58/__tests__/serialize.test.tsx#L151-L165

**Comment by hahnbeelee** [comment]
Hmm interesting

**Comment by hahnbeelee** [comment]
I think we're just going to try to migrate to mdx-bundler 🥲 Thank you for all of your help though!

**Comment by hahnbeelee** [comment]
Hello! I wanted to give an update here. Later we found out that there was content that resembled javascript like: `{{var_name_ex}}` this was purely content and the variable `var_name_ex` did not exist. Since we're working with customer generated content, in the end we decided to remove all javascript using a remark plugin since it's dangerous anyway. We have decided to continue to use `next-mdx-remote`!
